### PR TITLE
fix issue #I8B8SV, #I7RZSB, and bug of exporting visual results

### PR DESCRIPTION
**Pull request by panshaowu**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)
1. fix issue [#I8B8SV](https://e.gitee.com/mind_spore/projects/67813/tasks/list?issue=I8B8SV) which was caused by API changes of MindSpore r2.2.0
2. fix issue [#I7RZSB](https://gitee.com/mindspore-lab/mindocr/issues/I7RZSB) by correcting document bugs
3. fix bugs of exporting visual results when the text direction is vertical

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Review by CaitinZhao — APPROVED** [review]

File: mindocr/utils/misc.py [hunk 1]
```diff
@@ -35,10 +35,18 @@ def fetch_optimizer_lr(opt):
...
+                    if ms.version.__version__ < "2.2"
+                    else learning_rate(opt.global_step).reshape(())
+                )
                 lr += (cur_dynamic_lr,)
         else:
-            lr = opt.learning_rate(opt.global_step - 1).reshape(())
+            lr = (
+                opt.learning_rate(opt.global_step - 1).reshape(())
+                if ms.version.__version__ < "2.2"
+                else opt.learning_rate(opt.global_step).reshape(())
```
Comment (SamitHuang) [inline mindocr/utils/misc.py:48]: should keep `opt.global_step - 1` for version >= 2.2, since the learning rate list index starts from 0

File: mindocr/optim/adan.py [hunk 2]
```diff
@@ -163,6 +163,8 @@ def construct(self, gradients):
         gradients = self.scale_grad(gradients)
         gradients = self._grad_sparse_indices_deduplicate(gradients)
         lr = self.get_lr()
+        if version.__version__ >= "2.2" and self._is_dynamic_lr_or_weight_decay():
```
Comment (SamitHuang) [inline mindocr/optim/adan.py:166]: `self._is_dynamic_lr_or_weight_decay()` this condition is not necessary. No matter dynamic lr or constant lr, we should increase global_step in each optimization step.

Comment (panshaowu) [reply mindocr/optim/adan.py:166]: The PR has been revised according to this review comments. The condition `self._is_dynamic_lr_or_weight_decay()` is unnecessary because `self.global_step` is only used for `is_dynamic_lr` and `is_weight_decay` scenarios.

File: mindocr/utils/misc.py (hunk 1 above)
Comment (panshaowu) [reply mindocr/utils/misc.py]: The PR has been revised according to this review comments. The attribute `self.global_step` does not increase automatically for MindSpore >= r2.2.0, So the codes above was added to avoid IndexError. But it is unnecessary  because `self.global_step` += 1 has been added to the optimizer classes.

**Review by CaitinZhao — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

**Review by tonytonglt — APPROVED** [review]
